### PR TITLE
fix: issues while processing donations

### PR DESCRIPTION
**Pull request by mehul0810**
## Description
This PR is to fix the security issues reported.

## How Has This Been Tested?
I've tested this PR with the below points:

- [x] It is easy to SPAM Give with Pending donations, bypassing Akismet.
- [x] During donation processing you also allow wordpress user creation.
- [x] Reset all nonces via ajax call

## Types of changes
Bug fix (non-breaking change which fixes an issue) 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/process-donation.php [hunk 1]
```diff
@@ -29,7 +29,27 @@
...
+	}
+
+	$form_id  = ! empty( $post_data['give-form-id'] ) ? intval( $post_data['give-form-id'] ) : false;
+	$form_url = get_the_permalink( $form_id );
+
+	// Ensure that form url via form id matches form url via form submission.
+	if ( $form_url !== $post_data['give-form-url'] ) {
+		give_set_error(
+			'give-donation-form-spam-submission-error',
+			__( 'Our system has detected that you\'ve manipulated donation form details to submit spam.', 'give' )
```
Comment (DevinWalker) [inline includes/process-donation.php]: Fix typo: 

> Our system has detected that you have manipulated the donation form details.

**Comment by mehul0810** [comment]
Closing this PR. I've communicated for the same in Google doc.

**Comment by denisjacobi** [comment]
I'm unsure how effective these changes would really be to prevent a spammer :smirk:

Isn't the http referer just set by the client as part of the http request header?
The give-form-id and give-form-url are easily learned from the html page.

So wouldn't a spammer just need look at a donation page and then set those parameters as part of their http request header and data?

**Comment by ravinderk** [comment]
@denisjacobi Thast why we closed this PR. We are working on this.

**Pull request closed without merging**